src/API/AbstractAPIClient.php

```diff
@@ -41,25 +41,17 @@ public function callAPI(Request $request)
 
             $method = $request->getMethod();
             $params = $request->getParameters();
-            $url = $request->getUrl();
-            $headers = $request->getHeaders();
-            $body = $request->getBody();
-
-            $isPaginatable = false;
-            if ($method === 'GET') {
-                $isPaginatable = true;
-            }
 
             $mergedResponse = null;
 
             $currentPage = 1;
             $totalPages = 1;
 
             while ($totalPages >= $currentPage) {
```

Comment:
Why are we looping BEFORE we send the request?  Shouldn't all this while logic go inside `mergeResponses()` when `if (isset($response['result_info']))` fires?